### ISSUE TITLE
Semantics link:true does not work on web. 

### ISSUE THREAD

**Issue by jiahaog** [flutter/flutter#134795]
Internal Bug: b/300247612

### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

This can be reproduced on dartpad with the code sample below.

### Expected results

Wrapping a widget with
```
Semantics(
  link: true,
  child: Foo(),
)
```
should be labeled as a link. 

### Actual results

Instead, it's just labeled as a group.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

const Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(
        scaffoldBackgroundColor: darkBlue,
      ),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: MyWidget(),
        ),
      ),
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Semantics(
      link: true,
      child: Text(
        'Hello, World!',
        style: Theme.of(context).textTheme.headlineMedium,
      ),
    );
  }
}
```

</details>

### Flutter Doctor output

This can be reproduced on dartpad Based on Flutter 3.13.4 Dart SDK 3.1.2

**Comment by jiahaog** [comment]
P0 maps to the priority of the internal bug provided by the customer.

**Comment by chunhtai** [comment]
I think the internal bug is translate 1 to 1 per new triage process. cc @Hixie 

I just check the web engine, it doesn't seem like the SemanticsNode.isLink is used anywhere. It should probably build an anchor tag if isLink is true

**Comment by chunhtai** [comment]
also cc @yjbanov

**Comment by jiahaog** [comment]
> I think the internal bug is translate 1 to 1 per new triage process.

Please readjust the priorities I'm mistaken, but according to the mapping (recently updated by Ian in cl/553231427), the internal priority maps to P0 here though.

**Comment by math1man** [comment]
This is probably a dupe of https://github.com/flutter/flutter/issues/117011
